### DISCUSSION TITLE
Non-trivial example of binding to the DataTemplate of a ListView or CollectionView

### DISCUSSION THREAD

**Discussion by h0lg** [CommunityToolkit/Maui.Markup#236] [Q&A] [answered]
I'd like to **bind a collection of complex items** on a view model to the `DataTemplate` of a `ListView `or `CollectionView`.
I've found [this trivial example binding a list of flat items](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/markup/extensions/itemsview-extensions#itemtemplate), but don't know how to approach **binding item properties to different controls within** the `DataTemplate`.

Is there an example for that somewhere? If not - does anybody have an idea on how to do that? Thanks!

For example, how would I bind the `Blog.Posts` of the following view model to a `ListView `or `CollectionView`?

```c#
public class Blog
{
    public List<Post> Posts { get; set; }

    public class Post
    {
        public string Title { get; set; }
        public string Author { get; set; }
        public DateTime Published { get; set; }
    }
}
```

**Comment by TheCodeTraveler** [comment, marked as answer]
### Existing Examples from Sample App + Unit Tests

Here's a non-trivial example from our sample app: https://github.com/CommunityToolkit/Maui.Markup/blob/5ab559f471672ad78333a93c243a1180bd670c00/samples/CommunityToolkit.Maui.Markup.Sample/Pages/NewsPage.cs#L25

For your specific example, you'll need to use nested-bindings which are egregiously more complex. I highly recommend avoid using nested classes in your models to avoid nested-bindings.

We have a few examples of nested-bindings in our Unit Tests; here's one for example:

https://github.com/CommunityToolkit/Maui.Markup/blob/5ab559f471672ad78333a93c243a1180bd670c00/src/CommunityToolkit.Maui.Markup.UnitTests/TypedBindingExtensionsTests.cs#L297-L312

**Comment by h0lg** [comment]
Thank you for your prompt and detailed reply! I understand that 
1. [implementing a custom `DataTemplate`](https://github.com/CommunityToolkit/Maui.Markup/blob/5ab559f471672ad78333a93c243a1180bd670c00/samples/CommunityToolkit.Maui.Markup.Sample/Views/News/StoryDataTemplate.cs) is the intended way to bind properties of complex list items and
2. my specific example using nested view models would require complicated nested bindings, which luckily I can avoid.

That helps - I'll try that. Cheers!
